internal/verify/verifier.go

```diff
@@ -0,0 +1,49 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/fluxcd/source-controller/internal/verify/cosign"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/sigstore/cosign/pkg/oci"
+)
+
+// Enforce that Cosign implements Verifier interface.
+var _ Verifier = (*cosign.Verifier)(nil)
+
+// Verifier is a generic interface for verifiers.
+type Verifier interface {
```

Comment:
I find the naming of this package and/or interface too generic for the context of controller as a whole. E.g. verifying Git commit signatures is another thing for which we would use the name "verification" and/or "verify", and there is nothing that distinguishes this (cosign) from Git.

Reply:
I've updated the code according to your reviews.

Reply:
PTAL @Dentrax